engine/src/flutter/impeller/renderer/backend/metal/allocator_mtl.mm

```diff
@@ -19,75 +19,23 @@
 static bool DeviceSupportsDeviceTransientTargets(id<MTLDevice> device) {
   // Refer to the "Memoryless render targets" feature in the table below:
   // https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf
-  if (@available(ios 13.0, tvos 13.0, macos 10.15, *)) {
-    return [device supportsFamily:MTLGPUFamilyApple2];
-  } else {
-#if FML_OS_IOS
-    // This is perhaps redundant. But, just in case we somehow get into a case
-    // where Impeller runs on iOS versions less than 8.0 and/or without A8
-    // GPUs, we explicitly check feature set support.
-    return [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily2_v1];
-#else
-    // MacOS devices with Apple GPUs are only available with macos 10.15 and
-    // above. So, if we are here, it is safe to assume that memory-less targets
-    // are not supported.
-    return false;
-#endif
-  }
-  FML_UNREACHABLE();
-}
-
-static bool DeviceHasUnifiedMemoryArchitecture(id<MTLDevice> device) {
-  if (@available(ios 13.0, tvos 13.0, macOS 10.15, *)) {
-    return [device hasUnifiedMemory];
-  } else {
-#if FML_OS_IOS
-    // iOS devices where the availability check can fail always have had UMA.
-    return true;
-#else
-    // Mac devices where the availability check can fail have never had UMA.
-    return false;
-#endif
-  }
-  FML_UNREACHABLE();
+  return [device supportsFamily:MTLGPUFamilyApple2];
 }
 
 ISize DeviceMaxTextureSizeSupported(id<MTLDevice> device) {
-  // Since Apple didn't expose API for us to get the max texture size, we have
```

Comment:
Maybe keep this comment in? Since it lists the resource we got the numbers from. Otherwise, these numbers seem arbitrary.